### ISSUE TITLE
Windows: HMR doesn't work for a fresh Next.js project

### ISSUE THREAD

**Issue by BorisZubchenko** [chakra-ui/panda#1888]
### Description

Creating a fresh Next.js with PandaCSS:
1. I receive a warning on compilation. It may be Windows only. There is a separate issue: #1783.
2. Styles don't recompile. So I have to reboot the dev server every time I change styles.

### Link to Reproduction

Windows only?

### Steps to reproduce

1. Install via https://panda-css.com/docs/installation/nextjs
2. `pnpm dev`
3. Go to localhost:3000
4. Receive the following warning:
```
> next dev

   ▲ Next.js 14.0.4
   - Local:        http://localhost:3000

 ✓ Ready in 4.2s
 ○ Compiling / ...
🐼 info [hrtime] Extracted in (8.99ms)
 ⚠ ./app/globals.css.webpack[javascript/auto]!=!./node_modules/.pnpm/next@14.0.4_react-dom@18.2.0_react@18.2.0/node_modules/next/dist/build/webpack/loaders/css-loader/src/index.js??ruleSet[1].rules[11].oneOf[12].use[2]!./node_modules/.pnpm/next@14.0.4_react-dom@18.2.0_react@18.2.0/node_modules/next/dist/build/webpack/loaders/postcss-loader/src/index.js??ruleSet[1].rules[11].oneOf[12].use[3]!./app/globals.css
Invalid dependencies have been reported by plugins or loaders for this module. All reported dependencies need to be absolute paths.
Invalid dependencies may lead to broken watching and caching.
As best effort we try to convert all invalid values to absolute paths and converting globs into context dependencies, but this is deprecated behavior.
Loaders: Pass absolute paths to this.addDependency (existing files), this.addMissingDependency (not existing files), and this.addContextDependency (directories).
Plugins: Pass absolute paths to fileDependencies (existing files), missingDependencies (not existing files), and contextDependencies (directories).
Globs: They are not supported. Pass absolute path to the directory as context dependencies.
The following invalid values have been reported:
 * "C:/Users/boris/Projects/panda-nextjs/app"
 * "C:/Users/boris/Projects/panda-nextjs/node_modules/.pnpm/@pandacss+dev@0.23.0_typescript@5.3.3/node_modules/@pandacss/dev/dist/index.mjs"
 * "C:/Users/boris/Projects/panda-nextjs/pages"
 * and more ...

Import trace for requested module:
./app/globals.css.webpack[javascript/auto]!=!./node_modules/.pnpm/next@14.0.4_react-dom@18.2.0_react@18.2.0/node_modules/next/dist/build/webpack/loaders/css-loader/src/index.js??ruleSet[1].rules[11].oneOf[12].use[2]!./node_modules/.pnpm/next@14.0.4_react-dom@18.2.0_react@18.2.0/node_modules/next/dist/build/webpack/loaders/postcss-loader/src/index.js??ruleSet[1].rules[11].oneOf[12].use[3]!./app/globals.css
./app/globals.css
 ⚠ ./app/globals.css.webpack[javascript/auto]!=!./node_modules/.pnpm/next@14.0.4_react-dom@18.2.0_react@18.2.0/node_modules/next/dist/build/webpack/loaders/css-loader/src/index.js??ruleSet[1].rules[11].oneOf[12].use[2]!./node_modules/.pnpm/next@14.0.4_react-dom@18.2.0_react@18.2.0/node_modules/next/dist/build/webpack/loaders/postcss-loader/src/index.js??ruleSet[1].rules[11].oneOf[12].use[3]!./app/globals.css
Invalid dependencies have been reported by plugins or loaders for this module. All reported dependencies need to be absolute paths.
Invalid dependencies may lead to broken watching and caching.
As best effort we try to convert all invalid values to absolute paths and converting globs into context dependencies, but this is deprecated behavior.
Loaders: Pass absolute paths to this.addDependency (existing files), this.addMissingDependency (not existing files), and this.addContextDependency (directories).
Plugins: Pass absolute paths to fileDependencies (existing files), missingDependencies (not existing files), and contextDependencies (directories).
Globs: They are not supported. Pass absolute path to the directory as context dependencies.
The following invalid values have been reported:
 * "C:/Users/boris/Projects/panda-nextjs/app"
 * "C:/Users/boris/Projects/panda-nextjs/node_modules/.pnpm/@pandacss+dev@0.23.0_typescript@5.3.3/node_modules/@pandacss/dev/dist/index.mjs"
 * "C:/Users/boris/Projects/panda-nextjs/pages"
 * and more ...
```
5. Also, go to `page.tsx`:
```tsx
import { css } from '@/styled-system/css'

export default function Home() {
  return (
    <main className={css({bg: 'red'})}>Hello</main>
  )
}
```
6. Change `red` to `green`.
7. Panda still bundles the old color `red`. Since `green` wasn't bundled, the background is now white.
8. To make it work, I have to `pnpm dev` again.

### JS Framework

React (TS)

### Panda CSS Version

0.23.0

### Browser

N/A

### Operating System

- [ ] macOS
- [X] Windows
- [ ] Linux

**Comment by astahmer** [comment]
this looks specific to webpack (css-loader) on windows only. 
I would love to help but without a way to minimal repro I can't do much

just to make sure this isnt a postcss issue, does the same thing also happen when using a file generated from the CLI ?
what about removing the @layer with https://www.npmjs.com/package/@csstools/postcss-cascade-layers ? maybe css-loader doesn't handle those ?

**Comment by BorisZubchenko** [comment]
Assuming that "a file" is a `styled-system/styles.css`, I don't have it. It is not being generated. I followed [this guide](https://panda-css.com/docs/installation/cli). Maybe there's some configuration option to enable?

Regarding `@layer`, I will update the comment as soon as I reproduce it.

**Comment by astahmer** [comment]
> Assuming that "a file" is a styled-system/styles.css, I don't have it

it is ! it's generated when running the `pnpm panda` or `pnpm panda cssgen` command

**Comment by BorisZubchenko** [comment]
1. `@layer`: adding `@csstools/postcss-cascade-layers` had no effect. I also removed the `.next` folder and run `panda codegen --clean`.
2. `styles.css`: It works! I run it in a watch mode and everything's fine. No warnings.

**Comment by astahmer** [comment]
got it, so this would be a postcss issue, that helps a lot tracking the root cause 🙏

**Comment by MeanBoyCousin** [comment]
@astahmer 

I've also encountered this issue today after doing some dependency updates on my project. I've just run a bisect and from what I can see:
- 0.19.0 has no warnings in the terminal like in the OP.
- 0.21.0 has the warnings but still generates the global/reset/static files.
- 0.22.0 has the warnings and does not generate any of the files. Interestingly, using `cssgen -w` doesn't work for me.

Also running Next 14.0.4. My repo is private but happy to try and produce an MRE if needed.

**Comment by astahmer** [comment]
I tried something to fix it [here](https://github.com/chakra-ui/panda/commit/bc154358ebbd91b0a52aa4e3366f520829669c12#diff-042812ea1400639cf6d68f49fe599cc99fc48b32d130311c67f88ef85c9837caR21) but honestly I'm blindly trying as I don't have a windows machine, it would be very helpful to have someone with one that can investigate, it's most likely somewhere near that line

**Comment by BorisZubchenko** [comment]
For now, the error is the same. What I've done:
1. Cloned the repo, right after your commit
2. `pnpm install`
3. Went to `sandbox/next-js-app`
4. Run `pnpm prepare`, `pnpm dev`

If I had to do something else, could you please clarify? Meanwhile, I'll try to investigate the code itself.

```
PS S:\Projects\panda\sandbox\next-js-app> pnpm dev    

> sandbox-next-js-app@0.1.0 dev S:\Projects\panda\sandbox\next-js-app
> rimraf .next && next dev

   ▲ Next.js 14.0.3
   - Local:        http://localhost:3000

 ✓ Ready in 3.6s
 ○ Compiling / ...
🐼 info [hrtime] Extracted in (13.30ms)
 ⚠ ./src/app/globals.css.webpack[javascript/auto]!=!../../node_modules/.pnpm/next@14.0.3_@babel+core@7.12.9_react-dom@18.2.0_react@18.2.0/node_modules/next/dist/build/webpack/loaders/css-loader/src/inde
x.js??ruleSet[1].rules[12].oneOf[12].use[2]!../../node_modules/.pnpm/next@14.0.3_@babel+core@7.12.9_react-dom@18.2.0_react@18.2.0/node_modules/next/dist/build/webpack/loaders/postcss-loader/src/index.js??ruleSet[1].rules[12].oneOf[12].use[3]!./src/app/globals.css
Invalid dependencies have been reported by plugins or loaders for this module. All reported dependencies need to be absolute paths.
Invalid dependencies may lead to broken watching and caching.
As best effort we try to convert all invalid values to absolute paths and converting globs into context dependencies, but this is deprecated behavior.
Loaders: Pass absolute paths to this.addDependency (existing files), this.addMissingDependency (not existing files), and this.addContextDependency (directories).
Plugins: Pass absolute paths to fileDependencies (existing files), missingDependencies (not existing files), and contextDependencies (directories).
Globs: They are not supported. Pass absolute path to the directory as context dependencies.
The following invalid values have been reported:
 * "S:/Projects/panda/sandbox/next-js-app/pages"
 * "S:/Projects/panda/sandbox/next-js-app/panda.config.ts"
 * "S:/Projects/panda/sandbox/next-js-app/src"

Import trace for requested module:
./src/app/globals.css.webpack[javascript/auto]!=!../../node_modules/.pnpm/next@14.0.3_@babel+core@7.12.9_react-dom@18.2.0_react@18.2.0/node_modules/next/dist/build/webpack/loaders/css-loader/src/index.j
s??ruleSet[1].rules[12].oneOf[12].use[2]!../../node_modules/.pnpm/next@14.0.3_@babel+core@7.12.9_react-dom@18.2.0_react@18.2.0/node_modules/next/dist/build/webpack/loaders/postcss-loader/src/index.js??ruleSet[1].rules[12].oneOf[12].use[3]!./src/app/globals.css
./src/app/globals.css
```

**Comment by chrber04** [comment]
Experiencing same issue on Windows with 14.0.4 nextjs. Was driving me mad, so finding this thread was a relief.

Downgraded pandacss version to 0.19.0 "solved" it for now

**Comment by BorisZubchenko** [comment]
@chrber04, you can make it work on the latest version by running `panda --watch` in parallel with `next dev`:
```json
{ // package.json scripts
  "dev": "concurrently \"next dev\" \"panda --watch\"",
}
```
You need to clean the PostCSS config:
```js
const postcssConfig = {
  plugins: {},
}

module.exports = postcssConfig
```
Also, modify your `globals.css`:
```css
@import '../panda/system/styles.css'; // double-check the path, it depends on your configuration
```

**Comment by chrber04** [comment]
> @chrber04, you can make it work on the latest version by running `panda --watch` in parallel with `next dev`:
> 
> ```json
> { // package.json scripts
>   "dev": "concurrently \"next dev\" \"panda --watch\"",
> }
> ```
> 
> You need to clean the PostCSS config:
> 
> ```js
> const postcssConfig = {
>   plugins: {},
> }
> 
> module.exports = postcssConfig
> ```
> 
> Also, modify your `globals.css`:
> 
> ```css
> @import '../panda/system/styles.css'; /* double-check the path, it depends on your configuration */
> ```

Thank you so much, that did the job for me! And bonus points it seems to compile way faster now, which might just be the newer version but thanks all the same.

**Comment by astahmer** [comment]
sorry I should have made that clearer, this issue is specific to the `@pandacss/postcss` plugin so the [CLI](https://panda-css.com/docs/installation/cli) will always work just fine

> And bonus points it seems to compile way faster now,

yes ! the CLI will always be better performance-wise than the postcss plugin (because we don't control precisely how many times the postcss plugin will be called, it mostly depends on the host bundler (astro/vite/next etc))

**Comment by BorisZubchenko** [comment]
Is there any downside of not using the PostCSS plugin? Because if the CLI is faster and has the same behavior as with the plugin, I would choose the CLI.

**Comment by astahmer** [comment]
not really, the only downside is that you lose the convenience of only running 1 process (your dev command) and instead having to run dev + `pnpm panda -w`

**Comment by MeanBoyCousin** [comment]
> > @chrber04, you can make it work on the latest version by running `panda --watch` in parallel with `next dev`:
> > ```json
> > { // package.json scripts
> >   "dev": "concurrently \"next dev\" \"panda --watch\"",
> > }
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > You need to clean the PostCSS config:
> > ```js
> > const postcssConfig = {
> >   plugins: {},
> > }
> > 
> > module.exports = postcssConfig
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > Also, modify your `globals.css`:
> > ```css
> > @import '../panda/system/styles.css'; /* double-check the path, it depends on your configuration */
> > ```
> 
> Thank you so much, that did the job for me! And bonus points it seems to compile way faster now, which might just be the newer version but thanks all the same.

Thanks for this. Seems I was setting up wrong and this does indeed work.

Added tip for anyone who takes this route and is unfamiliar with `concurrently`, you can pass names to the commands to clean up your terminal as such: `"concurrently -n NextJS,Panda \"next dev\" \"panda --watch\""`

**Comment by omerman** [comment]
> not really, the only downside is that you lose the convenience of only running 1 process (your dev command) and instead having to run dev + `pnpm panda -w`

I would really love to get back on doing this 😅

**Comment by astahmer** [comment]
as mentioned previously, doing this will at least get the 2 processes running with 1 single command:

```json
{ // package.json scripts
  "dev": "concurrently \"next dev\" \"panda --watch\"",
}
```

**Comment by omerman** [comment]
Im actually using nx, Im using Mac and my partner uses Windows.
Up until HMR stopped working in dev.. everything was great.

here is a partial project.json of nx:

```
// This used to work well.
 "panda": {
      "command": "panda codegen --cwd apps/teachers-index",
      "defaultConfiguration": "production",
      "configurations": {
        "development": {},
        "test": {},
        "production": {}
      }
    },
    "dev": {
      "dependsOn": ["panda"],
      "executor": "@nx/next:server",
      "defaultConfiguration": "development",
      "configurations": {
        "development": {
          "buildTarget": "teachers-index:build:development",
          "dev": true,
          "port": 7778
        }
      }
    },
```
If you have ideas on how to make it easily work well on windows Im all ears

**Comment by neolectron** [comment]
Im using nx as well and concurently works fine with it, just update your scripts in package.json like you would normally do, nx does catch on this and allow you to run them.
You can also do this in project.json if you use this, or nx.json if you have one global config.

Hope it helps.

**Comment by omerman** [comment]
I'll be sure to check again thank you :)

**Comment by mewdriller** [comment]
I'm running into this issue on Windows as well and did a bit of digging through the various layers of `webpack` to see if I could find the root cause. It _looks_ like it's [right here in `NormalModule.js`](https://github.com/webpack/webpack/blob/main/lib/NormalModule.js#L97) where it's using a regex that doesn't allow for forward slashes in absolute paths even though they're valid in Windows.

From testing by hand-altering the files in my local `node_modules` a simple change to testing with `isAbsolute` from the `node:path` module would do the trick. However, there's a [comment](https://github.com/webpack/webpack/issues/13949#issuecomment-894826939) from Tobias on a [similar issue](https://github.com/webpack/webpack/issues/13949) that makes me think a PR wouldn't be easily accepted. 😅

**Comment by mewdriller** [comment]
OK, I did some more digging, it looks like we can prevent this issue by making a couple of alterations in `@pandacss/node`. The gist of the changes is that we want to change out usages of `pathe` for `node:path`.

It seems counterintuitive because `pathe` is supposed to make things normalized for Windows, but from the `webpack` perspective it's creating "invalid" absolute paths. Since both backslash and forward slash work work as path separators on windows it's safe to use the `webpack`-friendly backslash version.

I've made the changes in my local `node_modules` directly and things are working properly. I'll get started on a proper PR here in a bit.

**Comment by juliomuhlbauer** [comment]
Still not working

**Comment by mewdriller** [comment]
> Still not working

There was some [miscommunication](https://github.com/chakra-ui/panda/pull/1999#pullrequestreview-1822345354) on the merge/release handoff. It should be relatively easy to resolve.

**Comment by juliomuhlbauer** [comment]
This was fixed?

**Comment by anubra266** [comment]
Yes @juliomuhlbauer
